omigami/feature_selector.py

```diff
@@ -154,13 +162,22 @@ def create_outer_loop_results(
         return outer_loop_results
 
     def evaluate_min_mid_and_max_features(
-        self, input_data: InputDataset, best_features: SelectedFeatures, split: Split,
+        self,
+        input_data: InputDataset,
+        best_features: SelectedFeatures,
+        split: Split,
     ) -> Tuple[
         FeatureEvaluationResults, FeatureEvaluationResults, FeatureEvaluationResults
     ]:
-        outer_loop_data_min_feats = input_data.split_data(split)
-        outer_loop_data_max_feats = input_data.split_data(split)
-        outer_loop_data_mid_feats = input_data.split_data(split)
+        outer_loop_data_min_feats = self.data_splitter.split_data(
+            input_data, split, best_features.min_feats
+        )
+        outer_loop_data_max_feats = self.data_splitter.split_data(
+            input_data, split, best_features.max_feats
+        )
+        outer_loop_data_mid_feats = self.data_splitter.split_data(
+            input_data, split, best_features.mid_feats
+        )
```

Comment:
@giulio-tirabassi I think the bug might be fixed with this change

Reply:
well done!